### ISSUE TITLE
Fix LowerSETCC

### ISSUE THREAD

**Issue by krasin** [llvm-dcpu16/llvm-dcpu16#95]
Currently, LowerSETCC emits CMP and depends on Select_CC. It not only leads to a larger code size, (#78 and #93), but also to a crash of setcc.ll test, because too much of this lowering stuff is broken.

This issue is to implement DCPU16-specific LowerSET, in which the code:

```
< result > = seteq <ty> <var1>, <var2>
```

is lowered to something like

```
SET $reg, 0
IFE <var1> <var2>
SET $reg, 1
SET <result> , $reg
```

**Comment by krasin** [comment]
setcc.ll is broken in the following way:

```
$ bin/llc -march dcpu16 ../test/CodeGen/DCPU16/setcc.ll
llc: /disk2/llvm-dcpu16/lib/CodeGen/SelectionDAG/LegalizeDAG.cpp:702: void     {anonymous}::SelectionDAGLegalize::LegalizeOp(llvm::SDNode*): Assertion `TLI.getTypeAction(*DAG.getContext(),     Node->getValueType(i)) == TargetLowering::TypeLegal && "Unexpected illegal type!"' failed.
Stack dump:
0.  Program arguments: bin/llc -march dcpu16 ../test/CodeGen/DCPU16/setcc.ll 
1.  Running pass 'Function Pass Manager' on module '../test/CodeGen/DCPU16/setcc.ll'.
2.  Running pass 'DCPU16 DAG->DAG Pattern Instruction Selection' on function '@sccweqand'
Aborted
```

**Comment by krasin** [comment]
What I currently don't understand is how to properly introduce IFE (or IFG), because it has a specific that is nor branch, nor linear op.

**Comment by krasin** [comment]
I'm going to pretend that SETCC_{EQ, NE, GE} are the native instructions and define them in DCPU16InstInfo.td.

Later, we'll probably improve that, if needed.

**Comment by krasin** [comment]
I have started to play with InstInfo.td, but it's still incorrect, incomplete and under "cybernetical burst". The preview is 3e6b99ae2e5c9d00ffa4ee4eb68da967d4699e3b

Note: that code is incorrect in case of dst in [src1, src2].

**Comment by krasin** [comment]
Fiex by #109

**Comment by Blei** [comment]
Is this fixed or is still some work needed?

**Comment by ghost** [comment]
The setcc.ll was fixed by #106 / #109 . I think we can close this for now.

**Comment by krasin** [comment]
If you think that the issue is no longer valid, feel free to close it. :)
